cf/commands/domain/domains.go

```diff
@@ -30,66 +32,125 @@ func (cmd *ListDomains) MetaData() command_registry.CommandMetadata {
 	}
 }
 
-func (cmd *ListDomains) Requirements(requirementsFactory requirements.Factory, fc flags.FlagContext) (reqs []requirements.Requirement, err error) {
+func (cmd *ListDomains) Requirements(requirementsFactory requirements.Factory, fc flags.FlagContext) ([]requirements.Requirement, error) {
 	if len(fc.Args()) != 0 {
 		cmd.ui.Failed(T("Incorrect Usage. No argument required\n\n") + command_registry.Commands.CommandUsage("domains"))
 	}
 
-	cmd.orgReq = requirementsFactory.NewTargetedOrgRequirement()
-	reqs = []requirements.Requirement{
+	cmd.rountingApiRequirement = requirementsFactory.NewRoutingAPIRequirement()
+
+	reqs := []requirements.Requirement{
 		requirementsFactory.NewLoginRequirement(),
-		cmd.orgReq,
+		requirementsFactory.NewTargetedOrgRequirement(),
 	}
-	return
+
+	return reqs, nil
 }
 
 func (cmd *ListDomains) SetDependency(deps command_registry.Dependency, pluginCall bool) command_registry.Command {
 	cmd.ui = deps.Ui
 	cmd.config = deps.Config
 	cmd.domainRepo = deps.RepoLocator.GetDomainRepository()
+	cmd.routingApiRepo = deps.RepoLocator.GetRoutingApiRepository()
+
 	return cmd
 }
 
 func (cmd *ListDomains) Execute(c flags.FlagContext) {
-	org := cmd.orgReq.GetOrganizationFields()
+	org := cmd.config.OrganizationFields()
 
 	cmd.ui.Say(T("Getting domains in org {{.OrgName}} as {{.Username}}...",
 		map[string]interface{}{
 			"OrgName":  terminal.EntityNameColor(org.Name),
 			"Username": terminal.EntityNameColor(cmd.config.Username())}))
 
-	domains := cmd.fetchAllDomains(org.Guid)
-	cmd.printDomainsTable(domains)
+	domains, populateRouterGroups, err := cmd.getDomains(org.Guid)
+	if err != nil {
+		cmd.ui.Failed(T("Failed fetching domains.\n{{.Error}}", map[string]interface{}{"Error": err.Error()}))
+	}
 
 	if len(domains) == 0 {
 		cmd.ui.Say(T("No domains found"))
+		return
```

Comment:
This can't be added unless approved by Dies, but even then it's sort of a backwards-incompatible change because the command now behaves differently than it used to when it doesn't need to.